packages/devtools_app/lib/src/analytics_notification.dart

```diff
@@ -0,0 +1,196 @@
+  @override
+  Future<void> initialize() async {
+    analytics.exposeGaDevToolsEnabledToJs();
+    if (analytics.isGtagsReset()) {
+      await analytics.resetDevToolsFile();
+    }
+  }
+
+  bool _isEnabled;
+  @override
+  Future<bool> get isEnabled async => _isEnabled ??= await analytics.isEnabled;
+
+  bool _isFirstRun;
+  @override
+  Future<bool> get isFirstRun async =>
+      _isFirstRun ??= await analytics.isFirstRun;
+
+  @override
+  bool get isGtagsEnabled => analytics.isGtagsEnabled();
+
+  @override
+  void setAllowAnalytics() => platform.setAllowAnalytics();
+
+  @override
+  void setDontAllowAnalytics() => platform.setDontAllowAnalytics();
+
+  @override
+  void setUpAnalytics() {
+    analytics.initializeGA();
+    platform.jsHookupListenerForGA();
+  }
+}
+
+/// Conditionally displays a notification to request permission for collection
+/// of usage analytics.
+class AnalyticsNotification extends StatefulWidget {
+  const AnalyticsNotification({
+    @required this.provider,
+    @required this.child,
+  });
+
+  final Widget child;
+  final AnalyticsProvider provider;
+
+  @override
+  State<AnalyticsNotification> createState() =>
+      _AnalyticsNotificationState(provider, child);
+}
+
+class _AnalyticsNotificationState extends State<AnalyticsNotification> {
+  _AnalyticsNotificationState(this._provider, this._child);
+
+  final Widget _child;
+  final AnalyticsProvider _provider;
+
+  bool _isVisible = false;
+
+  @override
+  void initState() {
+    super.initState();
+    () async {
+      try {
+        await _provider.initialize();
+
+        if (_provider.isGtagsEnabled) {
+          if (await _provider.isFirstRun) {
+            setState(() {
+              _isVisible = true;
+            });
+          } else if (await _provider.isEnabled) {
+            _provider.setUpAnalytics();
+          }
+        }
+      } catch (e) {
+        print('Error collecting analytics:\n$e');
+      }
+    }();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+      children: [
+        Expanded(
+            child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            if (_isVisible)
+              Card(
+                margin: const EdgeInsets.only(bottom: denseRowSpacing),
+                child: Padding(
+                    padding: const EdgeInsets.all(defaultSpacing),
+                    child: Column(
+                      mainAxisSize: MainAxisSize.min,
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        Text(
+                          'Send usage statistics for DevTools?',
+                          style: textTheme.headline5,
+                        ),
+                        const Padding(padding: EdgeInsets.only(top: 20.0)),
```

Comment:
use defauiltSpacing const or add another const to theme.dart if we need 20.0 (e.g. wideSpacing or similar)